src/tools/check_brokenURL.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/python
+
+import re
+import os
+import sys
+import getopt
+import requests
+import argparse
+
+def get_filepaths(directory):
+    file_paths = []  
+    for root, directories, files in os.walk(directory):
+        for filename in files:
+            filepath = os.path.join(root, filename)
+            file_paths.append(filepath)
+    return file_paths
+
+def checkBrokenLink(url):
```

Comment:
we have to support `ignore links based on regex`. there are thousands of links that don't need to check. for example, we posted `https://github.com/ceph/ceph/pull/\d+` in our release note. it will take too much time to check those links.